autoload/sj/php.vim

```diff
@@ -212,6 +228,14 @@ function! sj#php#SplitMethodChain()
 
   let body = sj#GetCols(start_col, end_col)
   call sj#ReplaceCols(start_col, end_col, "\n".body)
+
+  if sj#settings#Read('php_method_chain_until_end_of_line')
+    call sj#PushCursor()
+    normal! j
+    call s:SplitNextArrow()
+    call sj#PopCursor()
+  endif
```

Comment:
You don't need to push and pop the cursor around the recursive call. Every top-level function is invoked in a push-pop bracket, so moving the cursor *inside* the function is not a problem, as long as it doesn't interfere with the function itself.